src/components/Registration/StartDate.vue

```diff
@@ -43,15 +43,11 @@ export default class StartDate extends Mixins(DateMixin) {
     startDateRef: DatePicker
   }
 
-  /** Initial date string value. */
-  @Prop({ default: '' })
-  readonly initialValue: string
-
   // Global actions
   @Action setRegistrationStartDate!: ActionBindingIF
 
   // Global getters
-  @Getter getCurrentJsDate!: Date
+  @Getter getRegistration!: RegistrationStateIF
```

Comment:
I figured, if this component writes to the store (line 47) then it might as well read from the store and save a prop and simplify the parent view.

Reply:
FYI, I'm also thinking about the layered architecture here:
- App.vue is the top level
- views (pages) are still high-level and shouldn't deal with "details" (if those details can be handled by components)
- components handle all the details, include store getters/actions if possible
- low-level (and shared/common) components should not read/write to the store, but should communicate with props/events if possible (as a last resort, they may have a callable public method)

The 2 ideas above are:
1. encapsulation of details / each file should know only what it needs to know
2. API to low-level components (which is part of 1. since low-level components should not know about the store)